esphome/components/lc709203f/lc709203f.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/i2c/i2c.h"
+
+namespace esphome {
+namespace lc709203f {
+
+static const uint8_t LC709203F_STATE_INIT = 0x01;
+static const uint8_t LC709203F_STATE_RSOC = 0x02;
+static const uint8_t LC709203F_STATE_TEMP_SETUP = 0x03;
+static const uint8_t LC709203F_STATE_NORMAL = 0x00;
+
+/// Enum listing allowable voltage settings for the LC709203F.
+enum LC709203FBatteryVoltage {
+  LC709203F_BATTERY_VOLTAGE_3_8 = 0x0000,
+  LC709203F_BATTERY_VOLTAGE_3_7 = 0x0001,
+};
+
+class Lc709203f : public sensor::Sensor, public PollingComponent, public i2c::I2CDevice {
+ public:
+  void setup() override;
+  void update() override;
+  void dump_config() override;
+
+  void set_pack_size(uint16_t pack_size);
+  void set_thermistor_b_constant(uint16_t b_constant);
+  void set_pack_voltage(LC709203FBatteryVoltage pack_voltage);
+  void set_voltage_sensor(sensor::Sensor *voltage_sensor) { voltage_sensor_ = voltage_sensor; }
+  void set_battery_remaining_sensor(sensor::Sensor *battery_remaining_sensor) {
+    battery_remaining_sensor_ = battery_remaining_sensor;
+  }
+  void set_temperature_sensor(sensor::Sensor *temperature_sensor) { temperature_sensor_ = temperature_sensor; }
```

Comment:
Please ensure you have the `this->` prefix on all member variable accesses & calls to class methods.
```suggestion
  void set_voltage_sensor(sensor::Sensor *voltage_sensor) { this->voltage_sensor_ = voltage_sensor; }
  void set_battery_remaining_sensor(sensor::Sensor *battery_remaining_sensor) {
    this->battery_remaining_sensor_ = battery_remaining_sensor;
  }
  void set_temperature_sensor(sensor::Sensor *temperature_sensor) { this->temperature_sensor_ = temperature_sensor; }
```